.circleci/wait-until-previous-builds-complete.sh

```diff
@@ -0,0 +1,165 @@
+#! /bin/bash
+
+set -e
+set -o pipefail
+set -o functrace
+
+#********************
+# VARIABLES
+# *******************
+api_root="https://circleci.com/api/v2/"
+# polling branch name. Other branches (PRs) are excluded from consideration.
+branch="master"
+# Polling workflow name. Other workflow names (nightly-tests and deploy-staging) are excluded from consideration.
+workflow_name="build-test-deploy"
+project_slug="gh/all-of-us"
+
+# Max check time on a workflow is 45 minutes because e2e tests may take a long time to finish.
+# DISCLAIMER This max time may not be enough if there are 2 or more workflows in waiting before this workflow.
+max_time=$((45 * 60))
+sleep_time=30
+
+#********************
+# FUNCTIONS
+# *******************
+
+# CIRCLECI_TOKEN is a env variable whose value is personal token.
+# See https://circleci.com/docs/2.0/managing-api-tokens/#creating-a-personal-api-token
+check_circleci_token() {
+  if [[ ! $CIRCLECI_TOKEN ]]; then
+    printf '%s\n' "Required env variable \"CIRCLECI_TOKEN\" is not found.\n Create a personal token then create \"CIRCLECI_TOKEN\" env variable?\n" >&2
+    exit 1
+  fi
+}
+
+post() {
```

Comment:
nit: would probably call these `circle_post`, `circle_get`; they specific to Circle